jcl/src/java.base/share/classes/java/security/AccessController.java

```diff
@@ -931,7 +940,15 @@ private static void checkPermsNPE(Permission... perms) {
 	checkPermsNPE(perms);
 	ProtectionDomain domain = getCallerPD(1);
 	ProtectionDomain[] pdArray = (domain == null) ? null : new ProtectionDomain[] { domain };
-	return doPrivileged(action, new AccessControlContext(context, pdArray, getNewAuthorizedState(context, domain)), perms);
+	AccessControlContext computedContext = getContextHelper(true);
+	AccessControlContext fixedContext = new AccessControlContext((context == null) ? computedContext : context,
+		pdArray, getNewAuthorizedState(context, domain));
+	if ((fixedContext.domainCombiner == null) && (computedContext.domainCombiner != null)) {
```

Comment:
Looking over this part, I'm not sure how correct it actually is. If we are given an actual `AccessControlContext`, should we replace the `domainCombiner` if it's null?

Reply:
Please run `sanity.functional`, `extended.functional` and internal compliance tests for Java 8/11/17/18.